server/src/test/java/io/crate/integrationtests/ViewsITest.java

```diff
@@ -303,4 +303,21 @@ public void test_object_column_of_view_contain_inner_types() throws Exception {
             "o['value_2']| integer"
         );
     }
+
+    /*
+     * https://github.com/crate/crate/issues/17836
+     */
+    @Test
+    public void test_error_on_unknown_object_key_on_authenticated_user() {
+        execute("CREATE TABLE doc.tbl1 (obj OBJECT(DYNAMIC))");
+        execute("CREATE USER user1");
+        execute("ALTER USER user1 SET (\"error_on_unknown_object_key\"=false)");
+        execute("GRANT ALL TO user1");
```

Comment:
I think this could be simplified to not use a custom user but just use a session and set the setting before the view generation and reset it before querying the view. At least to me, this would highlight the scenario in a better way (not being confused by the user creation). Also this would avoid using the `executeAs`. But no strong opinion.